packages/protocol/src/action-protocol/element-creation.ts

```diff
@@ -14,69 +14,159 @@
+
 /**
- * In order to create a node in the model the client can send a CreateNodeOperation with the necessary information to create that node.
+ * In order to create a node in the model the client can send a CreateNodeOperation with the necessary information to create
+ * the element that corresponds to that node in the source model.
+ * The corresponding namespace declares the action kind as constant and offers helper functions for type guard checks
+ * and creating new `CreateNodeOperations`.
  */
-export class CreateNodeOperation implements CreateOperation {
-    static readonly KIND = 'createNode';
-
-    constructor(
-        public readonly elementTypeId: string,
-        public location?: Point,
-        public containerId?: string,
-        public args?: Args,
-        public readonly kind: string = CreateNodeOperation.KIND
-    ) {}
-}
+export interface CreateNodeOperation extends CreateOperation {
+    /**
+     * The unique action kind.
+     */
+    kind: typeof CreateNodeOperation.KIND;
 
-export function isCreateNodeOperation(action: any): action is CreateNodeOperation {
-    return isActionKind(action, CreateNodeOperation.KIND) && isString(action, 'elementTypeId');
+    /*
+     * The location at which the operation shall be executed.
+     */
+    location?: Point;
+
+    /*
+     * The id of container element in which the node should be created. If not defined
+     * the root element will be used.
+     */
+    containerId?: string;
 }
 
+export namespace CreateNodeOperation {
+    export const KIND = 'createNode';
+
+    export function is(object: any): object is CreateNodeOperation {
+        return CreateOperation.hasKind(object, KIND);
+    }
+
+    export function create(
+        elementTypeId: string,
+        options: { location?: Point; containerId?: string; args?: Args } = {}
+    ): CreateNodeOperation {
+        return {
+            kind: KIND,
+            isOperation: true,
+            elementTypeId,
+            ...options
+        };
+    }
+}
 /**
- * In order to create an edge in the model the client can send a `CreateEdgeOperation` with the necessary information to create that edge.
+ * In order to create an edge in the model the client can send a `CreateEdgeOperation` with the necessary information to create
+ * the element that corresponds to that edge in the source model.
+ * The corresponding namespace declares the action kind as constant and offers helper functions for type guard checks
+ * and creating new `CreateEdgeOperations`.
  */
-export class CreateEdgeOperation implements CreateOperation {
-    static readonly KIND = 'createEdge';
-
-    constructor(
-        public readonly elementTypeId: string,
-        public sourceElementId: string,
-        public targetElementId: string,
-        public args?: Args,
-        public readonly kind: string = CreateEdgeOperation.KIND
-    ) {}
-}
+export interface CreateEdgeOperation extends CreateOperation {
+    /**
+     * The unique action kind.
+     */
```

Comment:
I'd spare such comments, also if it hurts to be "incomplete". But imho they just add noise and add no information.